### PR TITLE
Upgrade to Django 3.2 LTS and Dockerize junction

### LINKED ISSUE #611: Upgrade Django to 2.2 LTS
The things we'd need to do here are:

- [ ] Resolve the deprecation warnings.
- [ ] Upgrade to Django 1.11, and bump all dependencies. #664
- [ ] Upgrade to Django 2.2, and bump dependencies. #687

### PR DESCRIPTION
**Pull request by ananyo2012**
This PR aims to resolve the long pending junction upgrade to Django LTS version and support Python3 in production

We have also Dockerized junction for enabling cross platform setup and using docker for production deployment.

Thanks to @Bhandari423, @inovizz, @rohitchopra-epam for their efforts for their constant efforts and completing this gigantic task

Closes #611, #614, #730

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        nodejs \
+        npm \
+        libpq-dev && \
+    rm -rf /var/lib/apt/lists/*
+
+COPY requirements.txt /code/
+RUN pip install --no-cache-dir -r requirements.txt
+
+# Install requirements for running tests
+COPY ./tools/requirements-test.txt /code/
```
Comment (sayatul) [inline Dockerfile:19]: keeping test requirements in is not a good practice these should be installed while running on CI

Comment (inovizz) [reply Dockerfile:19]: Good point, I'll raise a fix over the weekend. Thanks.

File: settings/common.py [hunk 2]
```diff
@@ -159,13 +159,13 @@
...
-EMAIL_HOST_USER = DEFAULT_FROM_EMAIL = (os.environ.get("EMAIL_HOST_USER", "enter gmail id"),)
+EMAIL_HOST = "smtp.sendgrid.net"
+EMAIL_HOST_USER = os.environ.get("EMAIL_HOST_USER", "apikey")
 # turn on 2-step verification in your gmail account and add App password
-EMAIL_HOST_PASSWORD = (os.environ.get("EMAIL_HOST_PASSWORD", "enter App password"),)
+EMAIL_HOST_PASSWORD = os.environ.get("EMAIL_HOST_PASSWORD", "")
 EMAIL_PORT = 587
 EMAIL_USE_TLS = True
-# DEFAULT_FROM_EMAIL = SITE_VARIABLES["site_name"] + " <noreply@pssi.org.in>"
+DEFAULT_FROM_EMAIL = "pyconindia2023@gmail.com"
```
Comment (sayatul) [inline settings/common.py:168]: should be picked from env variable or settings, which will avoid this change every year

Comment (ananyo2012) [reply settings/common.py:168]: This will be replaced by official PyCon India email soon

**Comment by inovizz** [comment]
@ananyo2012 - now that we have seen issues while testing junction, maybe we could park this PR and merge only when all such issues are fixed?

**Comment by ananyo2012** [comment]
> @ananyo2012 - now that we have seen issues while testing junction, maybe we could park this PR and merge only when all such issues are fixed?

@inovizz Yes we can keep this on hold. But we do need to complete the upgrade soon so that others can consume the changes for junction upgrade. In the current stage we have to pretty much point everyone to this branch.

**Comment by inovizz** [comment]
@ananyo2012 - we could plan a sprint and complete open issues?

File: Dockerfile (hunk 1 above)
Comment (ananyo2012) [reply Dockerfile:19]: Let's keep this for now and move ahead. We will fix this while working on CI.

**Review by inovizz — APPROVED** [review]

**Comment by inovizz** [comment]
Great work @ananyo2012 as usual. 🚀

**Pull request merged**